circuit/src/threshold/mod.rs

```diff
@@ -1,2 +1,649 @@
+			let lt_eq_chipset =
+				LessEqualChipset::new(num_decomposed_limbs[i].clone(), max_limb_value.clone());
+			let res = lt_eq_chipset.synthesize(
+				&config.common,
+				&config.lt_eq,
+				layouter.namespace(|| "(num_decomposed) limb < max_limb_value"),
+			)?;
+			layouter.assign_region(
+				|| "res == 1",
+				|region| {
+					let mut ctx = RegionCtx::new(region, 0);
+					let res = ctx.copy_assign(config.common.advice[0], res.clone())?;
+					let one = ctx.copy_assign(config.common.advice[1], one.clone())?;
+					ctx.constrain_equal(res, one)?;
+
+					Ok(())
+				},
+			)?;
+		}
+
+		// check every element of "den_decomposed"
+		let den_decomposed_limbs = layouter.assign_region(
+			|| "den_decomposed",
+			|region| {
+				let mut limbs = vec![];
+
+				let mut ctx = RegionCtx::new(region, 0);
+				for i in 0..self.den_decomposed.len() {
+					let limb = ctx
+						.assign_advice(config.common.advice[i % ADVICE], self.den_decomposed[i])?;
+					limbs.push(limb);
+
+					if i % ADVICE == ADVICE - 1 {
+						ctx.next();
+					}
+				}
+				Ok(limbs)
+			},
+		)?;
+
+		for i in 0..den_decomposed_limbs.len() {
+			// assert!(limb < max_limb_value)
+			let lt_eq_chipset =
+				LessEqualChipset::new(den_decomposed_limbs[i].clone(), max_limb_value.clone());
+			let res = lt_eq_chipset.synthesize(
+				&config.common,
+				&config.lt_eq,
+				layouter.namespace(|| "(den_decomposed) limb < max_limb_value"),
+			)?;
+			layouter.assign_region(
+				|| "res == 1",
+				|region| {
+					let mut ctx = RegionCtx::new(region, 0);
+					let res = ctx.copy_assign(config.common.advice[0], res.clone())?;
+					let one = ctx.copy_assign(config.common.advice[1], one.clone())?;
+					ctx.constrain_equal(res, one)?;
+
+					Ok(())
+				},
+			)?;
+		}
+
+		// composed_num * composed_den_inv == score
+		let composed_num = {
+			let mut limbs = num_decomposed_limbs.clone();
+			limbs.reverse();
+			let scale = max_limb_value.clone();
+
+			let mut val = limbs[0].clone();
+			for limb in limbs.iter().take(NUM_LIMBS).skip(1) {
+				let mul_add_chipset = MulAddChipset::new(val, scale.clone(), limb.clone());
+				val = mul_add_chipset.synthesize(
+					&config.common,
+					&config.main,
+					layouter.namespace(|| "val * scale + limb"),
+				)?;
+			}
+
+			val
+		};
+
+		let composed_den = {
+			let mut limbs = den_decomposed_limbs.clone();
+			limbs.reverse();
+			let scale = max_limb_value;
+
+			let mut val = limbs[0].clone();
+			for limb in limbs.iter().take(NUM_LIMBS).skip(1) {
+				let mul_add_chipset = MulAddChipset::new(val, scale.clone(), limb.clone());
+				val = mul_add_chipset.synthesize(
+					&config.common,
+					&config.main,
+					layouter.namespace(|| "val * scale + limb"),
+				)?;
+			}
+
+			val
+		};
+
+		let composed_den_inv = {
+			let inv_chipset = InverseChipset::new(composed_den);
+			inv_chipset.synthesize(
+				&config.common,
+				&config.main,
+				layouter.namespace(|| "composed_den ^ -1"),
+			)?
+		};
```

Comment:
Also here the new scope is not needed

Reply:
Done https://github.com/eigen-trust/protocol/pull/316/commits/6fb493afaea8d9b5e40a7b86460dbb329bc149d0